### PR TITLE
Issue #SB-30227 fix: back-end user selection functionality

### PR DESCRIPTION
**Pull request by gouravmore**
Issue #SB-30227 fix: back-end user selection functionality

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -58,12 +58,39 @@ app.post(routes.API.TELEMMETRY, function (req, res) {
...
 });
 
+/**
+ * @param  {} routes.API.USER_ROLE This is the user role api url
+ * @param  {} res
+ * @param  {} {res.status(200) Sending the api response
+ * @param  {[]}} .json({userRoles}) Sending array response
+ */
+ app.get(routes.API.USER_ROLE, function (req, res) {
+  res.status(200).json(envVariables.USER_ROLE)
```
Comment (HarishGangula) [inline server/index.js:74]: Did you test with the environment variable?

File: server/index.js [hunk 2]
```diff
@@ -58,12 +58,39 @@ app.post(routes.API.TELEMMETRY, function (req, res) {
...
+});
+
+/**
+ * @param  {} routes.API.USERS This is the user role api url
+ * @param  {} res
+ * @param  {} {res.status(200) Sending the api response
+ * @param  {[]}} .json({userRoles}) Sending array response
+ */
+ app.post(routes.API.USERS, function (req, res) {
```
Comment (HarishGangula) [inline server/index.js:83]: Did you test with the environment variable?
what happens if the user role is not sent in the request body and what about validation of the request body for this API?

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 2 above)
Comment (gouravmore) [reply server/index.js:83]: If the roleType not sent in request body, empty users response will return.

Comment (HarishGangula) [reply server/index.js:83]: It should be standard way to validate as API should return proper status codes and handle other scenarios

File: server/index.js [hunk 3]
```diff
@@ -58,12 +61,63 @@ app.post(routes.API.TELEMMETRY, function (req, res) {
...
+ * @param  {} res
+ * @param  {} {res.status(200) Sending the api response
+ * @param  {[]}} .json({userRoles}) Sending array response
+ */
+ app.get(routes.API.USER_ROLE, function (req, res) {
+  let response = responseUtils.successResponse({
+    apiId: "api.v1.roles",
+    apiVersion: "1.0",
+    msgid: uuid(),
+    result: envVariables.USER_ROLE});
```
Comment (HarishGangula) [inline server/index.js:81]: Still, this didn't work if the env variable is set, have you tested this API by setting up env?

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: server/config/environment.js [hunk 4]
```diff
@@ -5,6 +5,9 @@ let envVariables = {
   API_AUTH_TOKEN: env.knowlg_bb_portal_api_auth_token || "",
   USER_API_TOKEN: env.knowlg_bb_portal_user_api_token || "",
   CHANNEL_ID: env.knowlg_bb_portal_channel_id || "",
-  USER_ID: env.knowlg_bb_portal_user_id || ""
+  USER_ID: env.knowlg_bb_portal_user_id || "",
+  USER_ROLE: env.knowlg_bb_portal_user_role || ['creator', 'reviewer'],
+  CREATORS: env.knowlg_bb_portal_users_creator || [{userName:'Creator',userId:'123gjh313',channelId:'12345',userToken:'1233123123'}],
+  REVIEWERS: env.knowlg_bb_portal_users_reviewer || [{userName:'Reviewer',userId:'123gjh313',channelId:'12345',userToken:'1233123123'}]
```
Comment (vinukumar-vs) [inline server/config/environment.js:11]: Creator & Reviewers default data is same. Change the userId's & tokens.

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

**Pull request merged**